zephyr/esp32s3/CMakeLists.txt

```diff
@@ -119,32 +119,32 @@ if(CONFIG_SOC_SERIES_ESP32S3)
   zephyr_library_sources_ifdef(CONFIG_COUNTER_TMR_ESP32 ../../components/hal/timer_hal.c)
 
   zephyr_sources_ifdef(
-    CONFIG_I2C_ESP32
+    CONFIG_USE_ESP32_HAL_I2C
     ../../components/hal/i2c_hal_iram.c
     ../../components/hal/i2c_hal.c
     )
 
   zephyr_library_sources_ifdef(CONFIG_WDT_ESP32 ../../components/hal/wdt_hal_iram.c)
 
   zephyr_sources_ifdef(
-    CONFIG_PWM_LED_ESP32
+    CONFIG_USE_ESP32_HAL_LED_PWM
     ../../components/soc/esp32s3/ledc_periph.c
     ../../components/hal/ledc_hal_iram.c
     ../../components/hal/ledc_hal.c
```

Comment:
question: shouldn't soc/hal code be splitted (like e.g. STM32 LL/HAL layers)? I don't have knowledge to say here.

Reply:
Are you ok if we stick only with the new Kconfig entry? `hal_espressif` is really tied to esp-idf components and these kind of modification isn't accomplished in a snap.

Reply:
It depends if it later requires a re-write. if hal code using soc code? how does it work?